pastelnetwork/pastel
# [PSL-342] `IsValid` for `Buy` and `Trade` check `intendedFor` field

- added additional check to IsValid method for Buy and Trade tickets.
When 'intendedFor' field in Sell ticket is not empty - it should match the Pastel ID of the Buy ticket (Buyer Pastel ID).
Same check is added to Trade ticket.ticket-processor
- refactored checks in Sell ticket for valid-before/valid-after fields.
Used new enum class SELL_TICKET_STATE to check Sell ticket status (not defined, not active, active, expired, unavailable).UNAVAILABLE
- added new parameter 'intendedFor' to 'tickets register sell' rpc API:
tickets register sell "nft-txid" "price" "PastelID" "passphrase" [valid-after] [valid-before] [copy-number] ["address"] ["intendedFor"]
....
9. "intendedFor"   (string, optional) The PastelID of the intended recipient of the NFT (empty by default).
- added additional checks for valid-before/valid-after parameters in "tickets register sell" API (rpc_check_unsigned_param)
- gtest for rpc_check_unsigned_param
- python test for intendedFor feature in mn_tickets.py:
   - create, register and activate NFT ticket
   - register Sell ticket with intendedFor field
   - check that Buy ticket cannot be registered with a Pastel ID not matching intendedFor
   - register Buy ticket
   - check that Trade ticket cannot be registered with a Pastel ID not matching intendedFor
   - register Trade ticket
